### DISCUSSION TITLE
Windows Server 2019 64bit won't execute blocky

### DISCUSSION THREAD

**Discussion by DaBrondo** [0xERR0R/blocky#155] [Q&A] [answered]
Error received:

This app can't run on your PC
To find a version for your PC, check with the software publisher.

Latest 0.8 release AMD64 Windows build.

**Comment by DaBrondo** [comment]
On another note, I downloaded the source and tried to build it with
set GOOS=windows
set GOARCH=amd64

It fails with this error:

c:\go\src\blocky\server\server.go:6:2: no Go files in c:\go\src\blocky\docs

**Comment by 0xERR0R** [comment]
Just tested 0.8 on Windows 10 -> works fine. How did you start it? With cmd.exe?
Regarding you build question: there is a step necessary to generate some code, see build step in the Makefile: https://github.com/0xERR0R/blocky/blob/d4e4fab306aca77c3cc2a52b6d76561db723d092/Makefile#L20

The swag tool can be installed as defined here: https://github.com/0xERR0R/blocky/blob/d4e4fab306aca77c3cc2a52b6d76561db723d092/Makefile#L12

**Comment by DaBrondo** [comment]
I tried starting it from a command window (administrator):  Same error.

When running from Powershell I get a more detailed error:

Program 'blocky_v0.8_windows_amd64.exe' failed to run: The specified executable is not a valid application for this OS
platform.At line:1 char:1

This is Server 2019 (which is only 64bit).

**Comment by DaBrondo** [comment]
Also, now when trying to build on this machine I get a bit further but:

# blocky/resolver
c:\go\src\blocky\resolver\query_logging_resolver.go:46:22: undefined: unix.Access
c:\go\src\blocky\resolver\query_logging_resolver.go:46:43: undefined: unix.W_OK
c:\go\src\blocky\resolver\stats_resolver.go:89:25: undefined: syscall.SIGUSR2

**Comment by 0xERR0R** [comment]
Found some similar issue here: https://stackoverflow.com/questions/60625330/error-while-installing-kubectl-in-windows-10
Please try to start from current directory `./blocky_v0.8_windows_amd64.exe`

**Comment by DaBrondo** [comment]
Same error when starting from current dir ".\blocky.exe"

In CMD.exe window it also gives the error "Access is denied".  This is an Administrator window too.

**Comment by 0xERR0R** [comment, marked as answer]
Do you try to compile the latest version (master)? This was fixed in #59

**Comment by DaBrondo** [comment]
I have not yet.... fairly new to Go lang.  If I go get blocky will it pull master?

**Comment by 0xERR0R** [comment]
Just clone the git repo

**Comment by 0xERR0R** [comment]
Please try to add the directory to your PATH variable (or temporary copy the binary to a director in PATH)

**Comment by DaBrondo** [comment]
Master built!

Testing now.

**Comment by DaBrondo** [comment]
So, it builds and is running.  Appears that the https://hosts-file.net/ad_servers.txt is down (or gone).  Is there another good list to grab?

**Comment by 0xERR0R** [comment]
Some list source are documented here: https://github.com/0xERR0R/blocky/wiki/List-sources

**Comment by mapl** [comment]
SHA1/SHA256 checksum should be provided for every binary release to ensure integrity.

[https://goreleaser.com/customization/checksum/](url)
#### GoReleaser generates a project_1.0.0_checksums.txt file and uploads it with the release, so your users can validate if the downloaded files are correct.

**Comment by 0xERR0R** [comment]
@mapl Currently, I don't use goreleaser, but providing a checksum is a good idea

**Reply by 0xERR0R** [reply]
> @mapl Currently, I don't use goreleaser, but providing a checksum is a good idea

Since Release 0.14, checksum file is provided
